zeebe/protocol-test-util/src/test/java/io/camunda/zeebe/test/broker/protocol/ProtocolFactoryTest.java

```diff
@@ -211,6 +215,62 @@ private static Stream<ValueType> provideValueTypes() {
     return ValueTypeMapping.getAcceptedValueTypes().stream();
   }
 
+  @Test
+  void shouldPopulateNestedObjectsInImplRecordValue() {
+    // given - CommandDistributionRecord has AuthInfo which is not part of the
+    // CommandDistributionRecordValue interface
+
+    // when
+    final var record =
+        factory.generateRecordWithImplValue(
+            ValueType.COMMAND_DISTRIBUTION, UnaryOperator.identity());
+
+    // then
+    RecordAssert.assertThat(record).hasValueType(ValueType.COMMAND_DISTRIBUTION);
+
+    // Cast to implementation class to access AuthInfo
+    final var implValue = (CommandDistributionRecord) record.getValue();
+
+    // Verify AuthInfo is populated
+    final var authInfo = implValue.getAuthInfo();
+    assertThat(authInfo).isNotNull();
+    assertThat(authInfo.getFormat())
+        .isNotNull()
+        .isNotEqualTo(AuthInfo.AuthDataFormat.UNKNOWN); // Should be randomized
+    assertThat(authInfo.getAuthData()).isNotEmpty();
```

Comment:
🔧 would be nice to check what's the record generated for `CommandDistributionRecord.getCommandValue()`